### ISSUE TITLE
issues around usage of `global` keyword 

### ISSUE THREAD

**Issue by pilif** [vimeo/psalm#2683]
It seems like psalm doesn't properly support the usage of `global` to declare a global variable:

https://psalm.dev/r/d23b41c6f3

```php
<?php

$c = 0;

function foo() : void {
    global $c;
    $c++;
}

foo();
printf("%d", $c);
```

this sample has two false positives: Inside of `foo()` it complains about the increment of `$c` to be Unused which isn't true because the `printf()` in the global scope does make use of the increment.

It also complains in `printf()` because it thinks `$c` is mixed, but it can't be.
